### PR TITLE
Replace export label with progress bar

### LINKED ISSUE #421: Use progress bars where appropriate
Instead of, for example, 

Exporting........

Have a progress bar. Where we can’t predict when the task will complete, make it indeterminate:

https://docs.oracle.com/javase/8/docs/api/javax/swing/JProgressBar.html#setIndeterminate-boolean-

### PR DESCRIPTION
**Pull request by emajj**
Closes #421

#### What has been done to verify that this works as intended?
Tested on linux mint

#### Why is this the best possible solution? Were any other approaches considered?
Looks more user friendly

#### Are there any risks to merging this code? If so, what are they?
Non I could think of.

### CODE REVIEW

File: src/org/opendatakit/briefcase/ui/export/ExportPanel.java [hunk 1]
```diff
@@ -203,11 +203,6 @@ public void onFormStatusEvent(FormStatusEvent event) {
     updateForms();
   }
 
-  @EventSubscriber(eventClass = ExportProgressEvent.class)
```
Comment (ggalmazor) [inline src/org/opendatakit/briefcase/ui/export/ExportPanel.java]: This removal and the next addition looks like you just reordered the methods on the file. Let's revert this change to keep changes to a minimum, plz :)

File: src/org/opendatakit/briefcase/ui/export/ExportPanelForm.java [hunk 2]
```diff
@@ -163,14 +149,19 @@ public void refresh() {
...
+  public void showExportProgressBar() {
+    exportProgressBar.setIndeterminate(true);
+    exportProgressBar.setVisible(true);
   }
 
-  public void hideExporting() {
-    exportingLabel.setVisible(false);
-    exportingLabel.setText(EXPORTING_DOT_ETC);
+  public void setProgressBarToCompleted() {
+    exportProgressBar.setIndeterminate(false);
```
Comment (ggalmazor) [inline src/org/opendatakit/briefcase/ui/export/ExportPanelForm.java]: I would expect the progress bar to be hidden when there's no activity...

File: src/org/opendatakit/briefcase/ui/export/ExportPanelForm.form [hunk 3]
```diff
@@ -51,15 +51,6 @@
             <properties/>
             <border type="none"/>
             <children>
-              <component id="b7dc5" class="javax.swing.JLabel" binding="exportingLabel">
```
Comment (ggalmazor) [inline src/org/opendatakit/briefcase/ui/export/ExportPanelForm.form]: As we talked over on Slack, the layout needs some work, and remember to rebuild after making changes on the UI Designer ;)

**Comment by ggalmazor** [comment]
It's really weird that you can't add a progress bar and IntelliJ adds a scroll pane instead... I've pushed a branch to my fork with what I think would work: https://github.com/ggalmazor/briefcase/commit/6cc4ab90cadd3fa59254345fd796065ad2038dad

File: src/org/opendatakit/briefcase/ui/export/ExportPanel.java (hunk 1 above)
Comment (emajj) [reply src/org/opendatakit/briefcase/ui/export/ExportPanel.java]: Ok will fix that. I actually deleted the method and had a new one that was listening to ExportProgressPercentageEvent with name onExportProgressPercentageEvent but since you said the progress event is only useful for a single export I deleted the method and added this one back. Didn't realize I'd change the positions. Will undo it.

**Comment by emajj** [comment]
@ggalmazor I think I have resolved all comments on this. I was able to add the hspacer without it adding a JScrollPane https://github.com/opendatakit/briefcase/pull/434/commits/e9bc9586d3e248a6907b1f526d3d3853cf062f2c,

and I'm glad I didn't have to use your commit. Makes me feel competent :laughing:

**Comment by emajj** [comment]
@ggalmazor updated

File: src/org/opendatakit/briefcase/ui/export/ExportPanelForm.java [hunk 4]
```diff
@@ -163,14 +153,14 @@ public void refresh() {
...
-    exportingLabel.setVisible(true);
+  public void showExportProgressBar() {
+    exportProgressBar.setVisible(true);
   }
 
-  public void hideExporting() {
-    exportingLabel.setVisible(false);
-    exportingLabel.setText(EXPORTING_DOT_ETC);
+  public void hideExportProgressBar() {
+    exportProgressBar.setValue(100);
```
Comment (ggalmazor) [inline src/org/opendatakit/briefcase/ui/export/ExportPanelForm.java]: Maybe just `setVisible(false)`? I believe setting the value and the indeterminate properties won't have any effect here...

**Comment by emajj** [comment]
@ggalmazor I finished working on the other tabs. Please have a look when you have time :slightly_smiling_face:

**Comment by emajj** [comment]
Also the Uploading.../Downloading... labels on the other tabs where appearing on the left so I maintained that but I forgot to mention that when you ask that I make the progress bar for the exporting tab appear on the right.

**Comment by ggalmazor** [comment]
@opendatakit-bot unlabel "needs review"
@opendatakit-bot label "needs testing"

**Comment by emajj** [comment]
@ggalmazor thanks. On to the next issue.

**Comment by kkrawczyk123** [comment]
@emajj I think it's a super cool feature! 

But it's working only on Ubuntu and Windows, it doesn't move on MacOS, screenshots attached:
screenshot from 2018-04-20 13-53-18
screenshot from 2018-04-20 13-52-44

@opendatakit-bot unlabel "needs testing"

**Comment by ggalmazor** [comment]
Maybe it doesn't have to move. It's an "undetermined" progress bar, meaning we can't set it to any certain % of progress.

**Comment by kkrawczyk123** [comment]
But if it's not moving it doesn't suggest any progress or action :( On Ubuntu and Windows, it's dancing from left to right.

**Comment by ggalmazor** [comment]
If this is some limitation in Swing I'm afraid we won't be able to do anything... We need to confirm this first.

**Comment by emajj** [comment]
@kkrawczyk123 @ggalmazor I have looked at how JProgressBars behave on Mac. I have not actually tested briefcase on a Mac. My host system is Mac OS High Sierra and when I ran this code 
```java
JFrame jFrame = new JFrame();
JProgressBar jProgressBar = new JProgressBar();
jProgressBar.setIndeterminate(true);
jProgressBar.setVisible(true);
jProgressBar.setStringPainted(false);
jFrame.setSize(300, 300);
jFrame.add(jProgressBar);
jFrame.setVisible(true);
jFrame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
```

And I get the output below 
screen shot 2018-04-20 at 2 47 19 pm

It may seem it's not moving but if you look carefully, the density of the blue color changes indicating some progress.

For instance in the image you shared, the density is on the far right.

**Comment by ggalmazor** [comment]
@emajj, thanks, I was expecting something like what you describe.
@kkrawczyk123 could you confirm this, please?

**Comment by yanokwa** [comment]
@ggalmazor Can you file an issue to make sure we remember to vertically and horizontally line up the bars? Once we have that, I can merge this.

**Comment by ggalmazor** [comment]
@yanokwa, done! #460
